### DISCUSSION TITLE
Why does JUnit Jupiter aggregator artifact has JUnit Jupiter Engine as a runtime dependency?

### DISCUSSION THREAD

**Discussion by ghost** [junit-team/junit-framework#3574] [General]
According to the [documentation](https://junit.org/junit5/docs/current/user-guide/#dependency-metadata-junit-jupiter) JUnit Jupiter test engine implementation is only required at runtime, despite the fact that it has entities which can be used in the client code. For example the Extension interfaces - Before/AfterEachMethodAdapter.

**Comment by marcphilipp** [comment]
`Before`/`AfterEachMethodAdapter` are not meant to be referenced by client code. Maybe you mean `Before`/`AfterEachCallback` which are part of `jupiter-api`?
